Un tipo de cuchillos nuevos... Saben que significa?

SagaStars opened the pull request:
_Seudo-Port de bay, creditos para ellos de las sprites_
**SERA UN DUELO A MUERTE CON CUCHILLOS**
![image](https://user-images.githubusercontent.com/55407528/75817917-5f026680-5d6e-11ea-91c5-a9276e5c0482.png)

- Añade un nuevo tipo de cuchillo... Un cuchillo plegable, el cual vendra de dos maneras:
**Normales:** Vienen en 3 tipos que solo cambian la sprite: como una navaja suiza, uno rustico y otro como un butterfly. 12 de daño y se podran conseguir en el xtra con una moneda y 2 en maint. Pueden guardarse en el bolsillo.
**Combate:** Haran 20 de daño y solo se pueden obtener en la armeria gamma (Vienen 2). O como objeto de traidor (Valor de 2 TC, como la energy dager) se ve super gangster tho.
**_Puedes cambiarle el color con alt+click, solo funciona una vez_**
## Why It's Good For The Game
Tanto como por rol y combate enriqueze la experiencia de una ligera manera, deja mejor me explico en vez de hablar paja. El hecho de portar un cuchillo plegable le sube como un +5 de intimidación a tu pj, te deja expresar mucho mejor la agresividad o paranoia que traes y es una mucho mejor alternativa para amenzar (en terminos de rol) que usando un stun prod y un par de zipties. Los robos de ID para agarrar dinero seran mas roleables, ejem PELEA A MUERTE CON CUCHILLOS. Y cosas asi, ya podras rolear bien al **GUACHAFAS**

## Images of changes
![image](https://user-images.githubusercontent.com/55407528/75819860-ba822380-5d71-11ea-9e43-38f013b31538.png)
![image](https://user-images.githubusercontent.com/55407528/75821390-8c521300-5d74-11ea-8069-fc483304490c.png)
![image](https://user-images.githubusercontent.com/55407528/76994764-4032d100-6925-11ea-98c3-a95d5bce5ce6.png)



## Changelog
:cl:
add: Cuchillos plegables hell yeah, agarratelo en un xtra con una monedita.
add: 2 cuchillos de estos plegables a maint, siempre van a aparecer ahi.
/:cl:

SagaStars committed:
SERA UN DUELO A MUERTE CON CUCHILLOS

No puedo creer que el cuchillo del chef haga 10 de daño kek, el del minero hace 15 ASJLKF

SagaStars commented:
Como sea, planeo hacer que funcione para cirujias ghetto pero por el bien de evitar conflictos esperare  por el rework de cirujia de paradise.

SagaStars committed:
Ya no son tan faciles de conseguir, ahora necesitas usar monedas para acceder a ellas con el xtra!

SagaStars commented:
Ahi ta

Evankhell561 commented on code/HISPANIA/game/objects/items/folding.dm:
```diff
@@ -0,0 +1,190 @@
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50)
+	sharp = FALSE
+
+	var/backstab_bonus = 4
```
donde no veo que esta variable haga nada.

SagaStars replied:
My bad, planeaba añadir una mecanica para backstabbear pero me heche para atras luego

Evankhell561 commented on code/game/objects/items/weapons/kitchen.dm:
```diff
@@ -108,9 +108,9 @@
+	force = 12
 	w_class = WEIGHT_CLASS_SMALL
-	throwforce = 10
+	throwforce = 12
```
que tiene que ver esto con tu pr?

SagaStars replied:
Para nivelarlo con los cuchillos que acabo de poner

Evankhell561 replied:
Si eso quieres hacer entonces bajale el daño a tus nuevos chuchillos, no le subas el daño a algo viejo.

SagaStars replied:
Es que ya estaba suavecito de daño desde antes solo piensa en que una espada normal como las del chaplain hacen 15 de daño, al fin al cabo +2 daño es una chucha, solo ayuda para con mobs que terminan en cifras como 48 cosas asi. _(no se si lo dije muchas veces pero, EL CUCHILLO DEL MINERO HACE 15 REEEEE :panic:)_

SagaStars replied:
![image0-6](https://user-images.githubusercontent.com/55407528/76136138-a144d580-6004-11ea-86ab-f7c7cad8baf0.png)

Evankhell561 replied:
Así hay más variedad

SagaStars replied:
Que?

SagaStars replied:
POOOR FAVOR EVAN, SON 2 PUNTITOS AAAAAAAAAAAA

Evankhell561 commented on code/modules/clothing/spacesuits/miscellaneous.dm:
```diff
@@ -285,4 +285,3 @@
 	sprite_sheets = list(
 		"Vox" = 'icons/mob/species/vox/suit.dmi')
 	sprite_sheets_obj = null
-	
```
devuelve esto

SagaStars replied:
Oookaaay

Evankhell561 commented on code/modules/clothing/suits/jobs.dm:
```diff
@@ -142,9 +142,7 @@
 	permeability_coefficient = 0.50
 	body_parts_covered = UPPER_TORSO|LOWER_TORSO|ARMS
 	allowed = list (/obj/item/kitchen/knife)
-  
```
devuelve esta linea

SagaStars replied:
OOOKAAAY

Evankhell561 commented on code/modules/clothing/suits/jobs.dm:
```diff
@@ -142,9 +142,7 @@
 	allowed = list (/obj/item/kitchen/knife)
-  
 	flags_size = ONESIZEFITSALL
-	
```
y esta

SagaStars replied:
AAAAAAAAAAAAAAAA

Evankhell561 commented on code/HISPANIA/game/objects/items/folding.dm:
```diff
@@ -0,0 +1,190 @@
+	desc = "This is a master item - berate the admin or mapper who spawned this"
+	foldedforce = 20
+	throwforce = 20
+	origin_tech = "materials=3;combat=4"
```
está bien los materiales pero darle combate 4 a esto es demasiado. las armas lasers tienen combate 4. No deberia un chichillo tener la misma tecnologia de combate de un arma laser.

SagaStars replied:
Well, yo no decidi esto. Son los niveles de tecnologia de los cuchillos de combate que vienen en la shuttle del codigo gamma.

Evankhell561 replied:
entiendo. bien.

SagaStars committed:
fix?

- Un par de arreglos muuuy importantes.
- Aprovecho para ajustar que el folding knife cuando sea lanzado no haga 15 de daño si esta cerrado.

SagaStars committed:
NADIIEEEE VIO NADA

SagaStars committed:
Merge branch 'master' into foldingknifes

Evankhell561 commented on code/HISPANIA/game/objects/items/folding.dm:
```diff
@@ -0,0 +1,190 @@
+	options["Blackout"] = color = COLOR_GRAY20
+	options["Cancel"] = null
+
+/proc/overlay_image(icon,icon_state,color,flags)
```
Define esteproc a nivel de tus cuchillos. No lo hagas global.

SagaStars replied:
Understandable, aunque podria ser util globalmente y lo pongo en la carpeta de procs... hm, nah

SagaStars replied:
Ya lo hago exclusivo del cuchillo

Evankhell561 commented on code/modules/clothing/suits/jobs.dm:
```diff
@@ -157,8 +155,10 @@
+
 	flags_size = ONESIZEFITSALL
 
+
```
Regresa esta linea

Spircen replied:
¡Revisa el PR de la chamaca de nothing! Va a llorar uwu

SagaStars replied:
Evan, estas lineas estan malditas pero lo volvere a intentar

SagaStars committed:
Fix#1

SagaStars committed:
Merge branch 'master' into foldingknifes

SagaStars committed:
Fix#2 ADIOOOOOOS LINEAS MALDITAS

SagaStars committed:
listo jefecito

SagaStars committed:
Cambio de rutas.

si señor

SagaStars commented:
Jefecito, va a venir a ver?

SagaStars commented:
Respecto a las preguntas que me hicistes:
# - Que peso tienen? Su peso deberia reducirse cuando está enfundado y aumentar cuando no.
Son tiny, por lo cual caben en bolsillo. Y correcto cuando se abre aumenta su tamaño a normal, y cuando se cierra se encoje.
![image](https://user-images.githubusercontent.com/55407528/76686805-9dace200-65f4-11ea-95ab-89ce26351196.png)

# - Que se diferencian estos cuchillos de combate nuevos de los cuchillos de combate?
Ademas de que se puedan plegar... pues no mucho, pero como habian tan pocos cuchillos en la armeria gamma (Solo 3) y teniamos esta bella sprite, pues aproveche.

SagaStars commented:
> Ademas de que se puedan plegar... pues no mucho, pero como habian tan pocos cuchillos en la armeria gamma (Solo 3) y teniamos esta bella sprite, pues aproveche.

Se me olvido decirlo, puse 2 mas.

SagaStars committed:
Ya ta evan, merga esta vaina

- Los remueve del loadout y a cambio coloca 2 en cuchillos en maint.

SagaStars committed:
Revert "Ya ta evan, merga esta vaina"

This reverts commit 2fe10bb94c2c8547a2152cf9131c7fbff2a7aae1.

SagaStars committed:
AHORA SI, ya ta

Quita los cuchillitos del loadout y agrega 2 cuchillos a maint.

SagaStars committed:
Merge branch 'master' into foldingknifes

SagaStars committed:
Remueve 1 cuchillo de la armeria gamma + cambio de z2 de cyberiad a z2 de hispania

Evankhell561 commented on _maps/map_files/cyberiad/z2.dmm:
```diff
@@ -6067,6 +6067,9 @@
 /obj/item/kitchen/knife/combat,
+/obj/item/kitchen/knife/folding/combat/tacticool,
+/obj/item/kitchen/knife/folding/combat/tacticool,
+/obj/item/kitchen/knife/folding/combat/tacticool,
```
no usamos el z2 de cyberiad sino el de hispania, estos cambios no son necesarios. y preferiria que no loas hicieras.

SagaStars replied:
L i s t o

SagaStars committed:
Evan no mutiles mi pobrecito cuerposito

Evankhell561 commented:
cuando los cuchillos de combate tuyos están cerrados tienen el daño por lanzamiento en 20 y cuando están abiertos tienen el daño por lanzamiento en 15

Evankhell561 reviewed (changes requested):
estos dos cuchillos se supone hacen el mismo daño sin embargo no lo pareciera. uno es notablemente más grande que el otro y el otro a penas se ve. 
![image](https://user-images.githubusercontent.com/49106866/77241896-49df6180-6bcf-11ea-8f39-84d181b98976.png)
![image](https://user-images.githubusercontent.com/49106866/77241914-8612c200-6bcf-11ea-8885-3a7696963449.png)

SagaStars committed:
Bazinga tristongo Bazinga tristongo

- Se le baja el daño del cuchillo de la armeria gamma a 18.
- Ahora la trowforce tiene sentido-

Evankhell561 commented on code/HISPANIA/game/objects/items/folding.dm:
```diff
@@ -0,0 +1,193 @@
+
+/obj/item/kitchen/knife/folding/proc/update_force()
+	if(open)
+		sharp = 1
```
TRUE

Evankhell561 commented on code/HISPANIA/game/objects/items/folding.dm:
```diff
@@ -0,0 +1,193 @@
+	options["Lighter Gray"] = color = COLOR_WHITE
+	options["Cancel"] = null
+
+/obj/item/kitchen/knife/folding/combat //master obj
```
Usas este item en algún lado? Donde lo encuentro?
Si lo unico que hace es servir de objeto padre admin spawn olvídalo. Mete una de sus variantes funcionales acá.

SagaStars committed:
Listo, el padre ya no existe, victoria para los gamers.

Evankhell561 reviewed (changes requested):
pon el true, arregla el changelogy creo que ya quedo

SagaStars committed:
Mi alma sera liberada del santuario maldito.

Evankhell561 reviewed (changes requested):
El changelog.

Evankhell561 commented:
Margearia esto ahora pero no he testeado el tema de los colores. Y no sé si al dar altclick aparezca un tipo de menú dandote a elegir entre blanco y #fadfca por ejemplo. Si pasas ss de eso ayudaría.

SagaStars committed:
Deberia afeitarme la barba o no?

SagaStars commented:
> Margearia esto ahora pero no he testeado el tema de los colores. Y no sé si al dar altclick aparezca un tipo de menú dandote a elegir entre blanco y #fadfca por ejemplo. Si pasas ss de eso ayudaría.

Ahora se cambia con una spraycan a diferencia de darle con alt+click. Aqui ta' la imagen.
PD: Siempre aparecia con un color random.
![image](https://user-images.githubusercontent.com/55407528/77692790-a8b12c00-6f7d-11ea-89fd-5b4fae4a73c8.png)
